python/lsst/ctrl/bps/bps_utils.py

```diff
@@ -0,0 +1,124 @@
+# This file is part of ctrl_bps.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Core functionality of BPS
+"""
+
+import os
+import contextlib
+import logging
+from importlib import import_module
+from enum import Enum
+
+_LOG = logging.getLogger()
+
+
+class WhenToSaveQuantumGraphs(Enum):
+    """Values for when to save the job quantum graphs."""
+    QGRAPH = 1   # Must be using single_quantum_clustering algorithm.
+    TRANSFORM = 2
+    PREPARE = 3
+    SUBMIT = 4
+
+
+def dynamically_load(name):
+    """Import at runtime given name.
+
+    Late import allows BPS to keep package dependencies minimal.
+    Also allows users to use their own subclasses without requiring
+    changes to BPS code.
+
+    Parameters
+    ----------
+    name: `str`
+        Name of class or function to load.  Must include full module path.
+
+    Returns
+    -------
+    The specified class ready to be instantiated
+    """
+    _LOG.debug("Dynamically importing %s", name)
+    (from_name, import_name) = name.rsplit(".", 1)
+    mod = import_module(from_name)
+    return getattr(mod, import_name)
```

Comment:
you might consider using `lsst.utils.doImport` to reduce duplication?

Reply:
I agree that this should definitely be using doImport.